sharedbuffers/mapped_struct.py

```diff
@@ -2667,18 +2904,29 @@ def init(self, slot_types = None, slot_keys = None, alignment = 8, pack_buffer_s
 
     @cython.ccall
     @cython.locals(has_bitmap = cython.ulonglong, none_bitmap = cython.ulonglong, present_bitmap = cython.ulonglong,
-        i = int)
+        i = int, get_values = cython.bint)
     @cython.returns(tuple)
-    def _get_bitmaps(self, obj):
+    def _get_bitmaps(self, obj, get_values=False):
         has_bitmap = 0
         none_bitmap = 0
+        if get_values:
+            values = []
         for i,slot in enumerate(self.slot_keys):
-            if hasattr(obj, slot):
+            try:
+                val = getattr(obj, slot)
+            except AttributeError:
+                pass
+            else:
                 has_bitmap |= cython.cast(cython.ulonglong, 1) << i
-                if getattr(obj, slot, 0) is None:
+                if val is None:
                     none_bitmap |= cython.cast(cython.ulonglong, 1) << i
+                elif get_values:
+                    values.append(val)
         present_bitmap = has_bitmap & ~none_bitmap
-        return has_bitmap, none_bitmap, present_bitmap
+        if get_values:
+            return has_bitmap, none_bitmap, present_bitmap, values
+        else:
+            return has_bitmap, none_bitmap, present_bitmap
```

Comment:
shouldn't this return a list in its fourth return value so the unpacking in ```get_packable``` doesn't fail ?

Reply:
The whole point of `get_values` is to change the shape of the returned tuple